qutebrowser/misc/mhtml.py

```diff
@@ -0,0 +1,387 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Daniel Schadt
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utils for writing a MHTML file."""
+
+import functools
+import io
+import os
+import re
+
+import collections
+import uuid
+from email import policy, generator, encoders
+from email.mime import multipart
+
+from PyQt5.QtCore import QUrl
+
+from qutebrowser.browser import webelem
+from qutebrowser.utils import log, objreg, message
+
+
+_File = collections.namedtuple("_File",
+                               ["content", "content_type", "content_location",
+                                "transfer_encoding"])
+
+
+_CSS_URL_PATTERNS = [re.compile(x) for x in [
+    r"@import\s+'(?P<url>[^']+)'",
+    r'@import\s+"(?P<url>[^"]+)"',
+    r'''url\((?P<url>[^'"][^)]*)\)''',
+    r'url\("(?P<url>[^"]+)"\)',
+    r"url\('(?P<url>[^']+)'\)",
+]]
```

Comment:
Could you do a quick check with `cssutils` and see if the tests pass? If so, I'm somewhat inclined to include it... After all, parsing things like CSS with regex is something which seems to work, and then you can be sure there are a lot of edge-cases where it doesn't :wink: